docs/Client.md

```diff
@@ -0,0 +1,1011 @@
+      - [Example 1 - Pipeline Echo](#example-1---pipeline-echo)
+    - [`Client.request(options [, callback])`](#clientrequestoptions--callback)
+      - [Parameter: `RequestOptions`](#parameter-requestoptions)
+      - [Parameter: `ResponseData`](#parameter-responsedata)
+      - [Example 1 - Basic GET Request](#example-1---basic-get-request)
+      - [Example 2 - Aborting a request](#example-2---aborting-a-request)
+    - [`Client.stream(options, factory [, callback])`](#clientstreamoptions-factory--callback)
+      - [Parameter: `StreamFactoryData`](#parameter-streamfactorydata)
+      - [Parameter: `StreamData`](#parameter-streamdata)
+      - [Example 1 - Basic GET stream request](#example-1---basic-get-stream-request)
+      - [Example 2 - Stream to Fastify Response](#example-2---stream-to-fastify-response)
+    - [`Client.upgrade(options[, callback])`](#clientupgradeoptions-callback)
+      - [Parameter: `UpgradeOptions`](#parameter-upgradeoptions)
+      - [Parameter: `UpgradeData`](#parameter-upgradedata)
+      - [Example 1 - Basic Upgrade Request](#example-1---basic-upgrade-request)
+  - [Instance Properties](#instance-properties)
+    - [`Client.busy`](#clientbusy)
+    - [`Client.closed`](#clientclosed)
+    - [`Client.connected`](#clientconnected)
+    - [`Client.destroyed`](#clientdestroyed)
+    - [`Client.pending`](#clientpending)
+    - [`Client.pipelining`](#clientpipelining)
+    - [`Client.running`](#clientrunning)
+    - [`Client.size`](#clientsize)
+  - [Instance Events](#instance-events)
+    - [Event: `'connect'`](#event-connect)
+      - [Example - Client connect event](#example---client-connect-event)
+    - [Event: `'disconnect'`](#event-disconnect)
+      - [Example - Client disconnect event](#example---client-disconnect-event)
+    - [Event: `'drain'`](#event-drain)
+      - [Example - Client drain event](#example---client-drain-event)
+  - [Parameter: `UndiciHeaders`](#parameter-undiciheaders)
+    - [Example 1 - Object](#example-1---object)
+    - [Example 2 - Array](#example-2---array)
+
+## `new Client(url, [options])`
+
+Arguments:
+
+* **url** `URL | string` - It should only include the protocol, hostname, and port.
+* **options** `object` (optional)
+  * **bodyTimeout** `number | null` (optional) - Default: `30e3` -
+  * **headersTimeout** `number | null` (optional) - Default: `30e3` - Node.js v14+ - The amount of time the parser will wait to receive the complete HTTP headers. Defaults to 30 seconds.
+  * **keepAliveMaxTimeout** `number | null` (optional) - Default: `600e3` - The maximum allowed `idleTimeout` when overriden by *keep-alive* hints from the server. Defaults to 10 minutes.
+  * **keepAliveTimeout** `number | null` (optional) - Default: `4e3` - The timeout after which a socket without active requests will time out. Monitors time between activity on a connected socket. This value may be overriden by *keep-alive* hints from the server. Defaults to 4 seconds.
```

Comment:
Add reference to developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Keep-Alive#directives